src/ImageRunModal.jsx

```diff
@@ -902,8 +925,49 @@ export class ImageRunModal extends React.Component {
                                         isReadOnly={!this.state.cpuSharesConfigure}
                                         onChange={value => this.onValueChanged('cpuShares', parseInt(value))} />
                                 </Flex>
-                            </FormGroup>}
-
+                            </FormGroup>
+                        }
+                        {this.state.owner === systemOwner && this.props.podmanRestartAvailable &&
+                        <Grid hasGutter md={6} sm={3}>
+                            <GridItem>
+                                <FormGroup fieldId='run-image-dialog-restart-policy' label={_("Restart policy")}
+                          labelIcon={
+                              <Popover aria-label={_("Restart policy help")}
+                                enableFlip
+                                bodyContent={_("Restart policy to follow when containers exit.")}>
+                                  <button onClick={e => e.preventDefault()} className="pf-c-form__group-label-help">
+                                      <OutlinedQuestionCircleIcon />
+                                  </button>
+                              </Popover>
+                          }
+                                >
+                                    <FormSelect id="run-image-dialog-restart-policy"
+                              aria-label={_("Restart policy help")}
+                              value={dialogValues.restartPolicy}
+                              onChange={value => this.onValueChanged('restartPolicy', value)}>
+                                        <FormSelectOption value='no' key='no' label={_("No")} />
+                                        <FormSelectOption value='on-failure' key='on-failure' label={_("On failure")} />
+                                        <FormSelectOption value='always' key='always' label={_("Always")} />
+                                    </FormSelect>
+                                </FormGroup>
+                            </GridItem>
+                            {dialogValues.restartPolicy === "on-failure" &&
+                                <FormGroup fieldId='run-image-dialog-restart-retries'
+                                  label={_("Maximum retries")}>
+                                    <TextInput
+                              type="number"
+                              id="run-image-dialog-restart-retries"
+                              className="run-image-dialog-restart-retries"
+                              value={dialogValues.restartTries}
+                              step={1}
+                              min={1}
+                              max={5}
```

Comment:
I don't see any limit from podman side - so lets not introduce any arbitrary value ourselves.